### ISSUE TITLE
identify at the level of <rank>

### ISSUE THREAD

**Issue by mfrasca** [mfrasca/bauble.classic#9]
from a email conversation with Eric (working at the botanical garden in Utrecht and providing us scientific support). we posed a question and he answered this way:
Q: for some of the problematic cases like 'quesito', 'cardón', rodilla de viejo', Saskia sort of knew the genus, but we still wonder what to write about the species 
A: We identify to genus level in this case and if a species is not certain a Identification qualifier (like cf.) can be added in the verification record

this is not the way bauble works.

currently bauble forces us to identify at the level of species, always. 
when we can identify the plant at the level of species, this goes fine, but in other cases we have to insert fictional data in the database.

more generic:
- plant identified at (rank) level of genus:
 - we insert a fictional 'sp' species in the genus.

- plant identified at level of family:
 - we insert a fictional 'problematic' genus in the family, then a fictional 'sp' species in the new fictional genus.

more specific:
- plant identified at level of variety, form, subspecies:
 - we create a new species, repeating the information relative to the common parent species, bud indicating that the species isn't really a species but it is a taxon at a more specific rank.

the last case has already been observed also in bauble.webapp and it has been solved by renaming the 'species' table to 'taxon' (see Bauble/bauble.webapp#4), but this does not solve the need to repeat information when more specific information is available nor the need to insert fictional objects when the information available isn't specific enough.

**Comment by mfrasca** [comment]
this is getting higher priority after first contact with Quito Botanical Garden. their need is different, but the source of the problem within bauble is the same and we can solve both issues...
at the QBG they have a large collection of orchids. for Orchidaceae (but also for Leguminosae) it is common to have several steps between familia and genus, namely subfamilia, tribu, subtribu.
these are completely specified once you have the genus, so one could argue they are useless once you have the genus, yet they help the reader group the different genera together.

**Comment by mfrasca** [comment]
@RoDuth
>   (I get a little concerned that your desire to "flatten" the data structure with "taxon" for family/genus/species is a fairly significant break from any off the botanic record systems I have seen over the years, but, as long as it is easy enough to get the data out in a ITF2/ABCD/etc. compatible format, e.g. for PlantSearch updates, I guess its no big deal?)

funny way to say "little concerned", while I guess you're just plain scared.
am I planning to flatten the data structure?
I would like to acknowledge that family/genus/species (but also tribe and variety) are all taxa, each with a rank and each with a parent taxon.
so from my point of view then possibly yes I want to flatten things into one table.
from your point of view I plan to offer you more levels (ranks) so actually the opposite of flattening.

**Comment by mfrasca** [comment]
mentioned in Bauble/bauble.classic#211

**Comment by RoDuth** [comment]
Accepted, and I do see the reasoning, its just such a major deviation from anything else I've seen.  From my perspective it is a case of what is the data for and how do you use it... For me it is mainly just what is where, when was it planted and where did it come from.  The taxonomic/nomenclature details are not something I need at hand necessarily but that’s maybe because I don't have particularly large collections of any specific taxonomic groups.  Also, nomenclature is such a.... messy art that I just don't have the time (or need) to get bogged down in it.  While we have collections such as Zingiberales, Bromeliaceae, Orchidaceae, Gymnospermae, Ferns and allies, we can always get what we need from the binomial and family.  I'm no fan of "double handling" data that can be found elsewhere, especially when they can be in such a state of flux, so the binomial is the key for me, if I ever need the rest I can go looking for it.

For the issues you have raised regarding plants not yet identified I believe there are conventions in place, at least with the botanists I deal with here there are.  The use of "botanist tag" names, arbitrary names at whichever level of identification. Something like  `RUB sp. (IGC1033)` an unnamed plant likely from the family Rubiaceae or `Actephila sp. Koumala (I.G Champion 870)` (which has now been formally identified as `Actephila championiae` [see here](http://biodiversity.org.au/nsl/services/search?product=apc&tree.id=1133571&name=Actephila+championiae&inc._scientific=&inc.scientific=on&inc._cultivar=&max=100&display=apc&search=true)).  I even have a family in the database called `Unknown` which is used when this is not even known and a `GEN` genus also.  When it's an unknown variety etc. it is written as `Dianella caerulea (Finch Hatton)` or something like this.  This is the way we currently communicate, Bauble does allow it and we are not likely to change even if there were other ways of doing it so for me I fear that such a radical change can only bring me trouble!  But as I say as long as I can continue to do things the way I do now... no big deal.  Right?

**Comment by mfrasca** [comment]
> I even have a family in the database called Unknown which is used when this is not even known and a GEN genus also

at JBQ, for plants identified at level of family but unknown genus, I suggested entering a genus named like the family, but prefixed with a "Zzz-", missing the trailing 'e', and obviously belonging to the family. in this genus we have the likewise unknown species "sp". eg: »_Zzz-orchidacea_ sp«
I guess I would name your Unknown family "Zzz-plantaceae" :smile: , then the genus I would probably call Zzz-planta and the species (the one you can finally associate to an accession) »_Zzz-planta_ sp«

the reason for that leading zzz is lexicographic: 'unknown', or 'problematic' (my first guess at the Cuchubo garden 2 years ago) would go somewhere in the middle, messing up the group of better identified plants.

at JBQ they told me they would be happy being able to group things by subfamilies, tribes and subtribes (they focus on Orchidaceae), because some taxonomists specialize in such sections of that huge family. so it would be nice for them, they said, to be able to show a list of plants sorted by subtribe, for example.

the solution I have offered is based on the exporting facility: the template will add subtribe information to the genus, then tribe information to subtribes and subfamily information to tribes. it has to be hard coded in the template. the template is public and you can [have a look](https://github.com/Bauble/bauble.classic/blob/master/bauble/plugins/report/mako/example_accession.csv) at it.

**Comment by RoDuth** [comment]
The examples of "botanist tags" I provided above are common practice with the taxonomists in Australia, they are all "real world" examples and may be found in published reference material so we stick to this format.  When a formal identification turns up in one of the reliable sources (Qld Herb Recs, APNI, etc.) we switch to that and the "botanist tag names" become synonyms because you still may come across them in old literature.  I have never thought of the fact that they may end up in the middle lexicographically as an issue because we see them as valid binomials (until a better comes along) anyway.  I can see the point though.  
Does the JBQ collection not have similarly "botanist tag" named plants?  Who does there identifications?  Do they have a taxonomist either as staff or which they work with?  Do they voucher? Are we talking wild collected, yet to be identified plants or plants that have lost there names to time and poor record keeping?  Species or potentially hybrids/cultivars etc.?  My thought on it is that the Binomial is the key to all that other data, its the equivalent of the accession number and if you have that right Binomial then so much of the other data is fairly easily accessible by clicking the relevant "link" in the info pane... but then... I don't have the Orchid problem... thankfully!!  :laughing:   As I said we do have some of these sorts of collection for which it is common to use the intermediate groupings (Broms, Orchids, Cycads, etc.), I just haven't had a need for them and hence any need to keep them in the database.  Family I do use a lot but nothing much more than that.  I'm assuming that JBQ DO use these intermediate groupings.  Do you know what for?

**Comment by mfrasca** [comment]
@felipead87, @TatiJaramilloV, read the above comment, it's a lot of questions for either of you, or for Lucho, but he doesn't yet have an account here.

**Comment by mfrasca** [comment]
@RoDuth , where do you put this "botanist tag"? can you paste here some information from your database? or the screenshot of a view on your data?

**Comment by RoDuth** [comment]
@felipead87, @TatiJaramilloV I just re-read my questions above, sorry if they sound a little presumptuous, it definitely wasn't my intent. I am just interested in the differences of end users needs and uses.  Its more a matter of interest than anything.  When we surveyed gardens in Australia and New Zealand about their database needs and wants the differences could be quite large and I wonder if there may be similarities with some of those who responded to our survey?  
@mfrasca I don't have our complete database at hand right now but here are 2, not so great examples of how we use "botanist tags" for names
![screenshot from 2015-12-11 07 59 43](https://cloud.githubusercontent.com/assets/15710055/11730027/23a218c6-9fde-11e5-90f8-86d211b7502a.png)
The first is just the place holder for complete unknowns that have not been seen by a botanist yet.  The 2 accessions are 2 different ferns from a collection trip that we (or any of our local experts) could not ID to any level confidently.  We will grow these on until they are large enough to get material to back voucher with the Qld Herbarium.  Once identified they will be renamed in Bauble but `GEN sp.` will NOT be considered a synonym (although it will be recorded in a change note of course).  The second is a similar situation of a plant that we collected but have sent a voucher of to the Queensland Herbarium and this is the name it came back as.  It has obviously proven to be an unrecorded species and will at some point in the future get a formal ID but for now it is referred to as `RUB sp. nov (IGC1033)` in correspondence, the Qld Herbarium Records, etc..  When it does receive a name `RUB sp. nov (IGC1033)` WILL be recorded as a synonym.  Here is how it looks in the species editor.  Not ideal as we would normally records Author (the Queensland Herbarium Botanist that assessed our voucher and returned the ID) but as I said I only have a small subset of our data on hand right now (am at home).
![screenshot from 2015-12-11 08 01 39](https://cloud.githubusercontent.com/assets/15710055/11730841/58705e8c-9fe3-11e5-975f-1eb4b3144fa4.png)

Hope that makes sense.

**Comment by RoDuth** [comment]
@felipead87, @TatiJaramilloV I just re-read my questions above, sorry if they sound a little presumptuous, it definitely wasn't my intent. I am just interested in the differences of end users needs and uses.  Its more a matter of interest than anything.  When we surveyed gardens in Australia and New Zealand about their database needs and wants the differences could be quite large and I wonder if there may be similarities with some of those who responded to our survey?  
@mfrasca I don't have our complete database at hand right now but here are 2, not so great examples of how we use "botanist tags" for names
![screenshot from 2015-12-11 07 59 43](https://cloud.githubusercontent.com/assets/15710055/11730027/23a218c6-9fde-11e5-90f8-86d211b7502a.png)
The first is just the place holder for complete unknowns that have not been seen by a botanist yet.  The 2 accessions are 2 different ferns from a collection trip that we (or any of our local experts) could not ID to any level confidently.  We will grow these on until they are large enough to get material to back voucher with the Qld Herbarium.  Once identified they will be renamed in Bauble but `GEN sp.` will NOT be considered a synonym (although it will be recorded in a change note of course).  The second is a similar situation of a plant that we collected but have sent a voucher of to the Queensland Herbarium and this is the name it came back as.  It has obviously proven to be an unrecorded species and will at some point in the future get a formal ID but for now it is referred to as `RUB sp. nov (IGC1033)` in correspondence, the Qld Herbarium Records, etc..  When it does receive a name `RUB sp. nov (IGC1033)` WILL be recorded as a synonym.
